data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -390,6 +391,89 @@ public void getStateSyncCommittees_shouldRejectFarFutureEpoch() {
+    final List<BLSPublicKey> committeeKeys =
+        List.of(
+            stateValidators.get(1),
+            stateValidators.get(4),
+            stateValidators.get(6),
+            stateValidators.get(7),
+            stateValidators.get(9),
+            stateValidators.get(11));
+
+    final Set<String> validators =
+        Set.of(committeeKeys.get(0).toHexString(), committeeKeys.get(2).toHexString(), "4");
+    final Map<Integer, Integer> committeeIndices =
+        provider.getCommitteeIndices(committeeKeys, validators, state);
+
+    assertThat(committeeIndices).containsExactlyInAnyOrderEntriesOf(Map.of(0, 1, 1, 4, 2, 6));
+  }
+
+  @Test
+  public void getCommitteeIndices_withNoSpecifiedValidators() {
+    final Spec spec = TestSpecFactory.createMinimalAltair();
+    final DataStructureUtil data = new DataStructureUtil(spec);
+    final ChainDataProvider provider = setupAltairState();
+    final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state =
+        data.randomBeaconState(32);
+
+    final List<BLSPublicKey> stateValidators =
+        state.getValidators().stream().map(Validator::getPublicKey).collect(toList());
+    final List<BLSPublicKey> committeeKeys =
+        List.of(
+            stateValidators.get(1),
+            stateValidators.get(4),
+            stateValidators.get(6),
+            stateValidators.get(7),
+            stateValidators.get(9),
+            stateValidators.get(11));
+
+    final Map<Integer, Integer> committeeIndices =
+        provider.getCommitteeIndices(committeeKeys, Set.of(), state);
+
+    assertThat(committeeIndices)
+        .containsExactlyInAnyOrderEntriesOf(Map.of(0, 1, 1, 4, 2, 6, 3, 7, 4, 9, 5, 11));
+  }
+
+  @Test
+  public void getCommitteeIndices_withSpecifiedValidatorsNotInCommittee() {
+    final Spec spec = TestSpecFactory.createMinimalAltair();
+    final DataStructureUtil data = new DataStructureUtil(spec);
+    final ChainDataProvider provider = setupAltairState();
+    final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state =
+        data.randomBeaconState(32);
+
+    final List<BLSPublicKey> stateValidators =
+        state.getValidators().stream().map(Validator::getPublicKey).collect(toList());
+    final List<BLSPublicKey> committeeKeys =
+        List.of(
+            stateValidators.get(1),
+            stateValidators.get(4),
+            stateValidators.get(6),
+            stateValidators.get(7),
+            stateValidators.get(9),
+            stateValidators.get(11));
+
+    final Set<String> validators =
+        Set.of(
+            data.randomPublicKey().toHexString(), // Validator not in committee
+            "2", // Validator not in committee
+            committeeKeys.get(2).toHexString());
+    final Map<Integer, Integer> committeeIndices =
+        provider.getCommitteeIndices(committeeKeys, validators, state);
+
+    assertThat(committeeIndices).containsExactlyInAnyOrderEntriesOf(Map.of(2, 6));
```

Comment:
if the specified validator isn't in a committee, why do we get a map entry?

Reply:
because one of them is in the committee

Reply:
oh this might be way too many 2's in the example, i think you're also adding committee member 2... add committee member 7 or something random...

Reply:
I commented which ones were added that are not in the committee (and they were not added)

Reply:
sure, but 2 everywhere becomes confusing and your brain associates it...